src/presentational-components/shared/content-list-empty-state.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import {
+  Bullseye,
+  Button,
+  EmptyState,
+  EmptyStateIcon,
+  EmptyStateBody,
+  EmptyStateSecondaryActions,
+  Text,
+  TextContent,
+  TextVariants
+} from '@patternfly/react-core';
+
+const ContentListEmptyState = ({
+  title,
+  Icon,
+  description,
+  PrimaryAction,
+  renderDescription
+}) => (
+  <Bullseye className="empty-state">
+    <EmptyState>
+      <EmptyStateIcon icon={ Icon } />
+      <TextContent>
+        <Text component={ TextVariants.h1 }>
+          { title }
+        </Text>
+      </TextContent>
+      <EmptyStateBody>
+        { description }
+        { renderDescription() }
+      </EmptyStateBody>
+      <EmptyStateSecondaryActions>
+        { PrimaryAction && <PrimaryAction /> }
+      </EmptyStateSecondaryActions>
+    </EmptyState>
+  </Bullseye>
+);
+
+ContentListEmptyState.defaultProps = {
+  renderDescription: () => null
+};
+
+ContentListEmptyState.propTypes = {
+  title: PropTypes.string.isRequired,
+  Icon: PropTypes.node.isRequired,
+  description: PropTypes.string.isRequired,
+  PrimaryAction: PropTypes.node,
+  renderDescription: PropTypes.func
```

Comment:
is there any reason for `description` and `renderDescription`?

I don't see that `renderDescription` is used anywhere in the code.

Reply:
it is used to provide custom description for each component's empty state, for example: https://github.com/ManageIQ/catalog-ui/blob/290f5d426b43cc7b3e2e8f1776027a837a148e77/src/smart-components/platform/platforms.js#L79
this is for the gallery. 
The same is provided here for lists.